network/p2p/handler.go

```diff
@@ -63,28 +63,44 @@ func (NoOpHandler) CrossChainAppRequest(context.Context, ids.ID, time.Time, []by
 	return nil, nil
 }
 
+func NewValidatorHandler(
+	handler Handler,
+	validatorSet ValidatorSet,
+	log logging.Logger,
+) *ValidatorHandler {
+	return &ValidatorHandler{
+		handler:      handler,
+		validatorSet: validatorSet,
+		log:          log,
+	}
+}
+
 // ValidatorHandler drops messages from non-validators
 type ValidatorHandler struct {
-	Handler
-	ValidatorSet ValidatorSet
-	Log          logging.Logger
+	handler      Handler
+	validatorSet ValidatorSet
+	log          logging.Logger
 }
 
 func (v ValidatorHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) {
-	if !v.ValidatorSet.Has(ctx, nodeID) {
-		v.Log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
+	if !v.validatorSet.Has(ctx, nodeID) {
+		v.log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
```

Comment:
can we specify the reason why we dropped these messages?

Reply:
Done